Sources/Plasma/PubUtilLib/plPhysX/plGenericPhysical.cpp

```diff
@@ -0,0 +1,506 @@
+        }
+    }
+
+    return false;
+}
+
+void plPXPhysical::GetSyncState(hsPoint3& pos, hsQuat& rot, hsVector3& linV, hsVector3& angV)
+{
+    IGetPoseSim(pos, rot);
+    GetLinearVelocitySim(linV);
+    GetAngularVelocitySim(angV);
+}
+
+void plPXPhysical::SetSyncState(hsPoint3* pos, hsQuat* rot, hsVector3* linV, hsVector3* angV)
+{
+    bool isLoading = plNetClientApp::GetInstance()->IsLoadingInitialAgeState();
+    bool isFirstIn = plNetClientApp::GetInstance()->GetJoinOrder() == 0;
+    bool initialSync = isLoading && isFirstIn;
+    bool wakeup = !(initialSync && GetProperty(plSimulationInterface::kStartInactive));
+
+    // If the physical has fallen out of the sim, and this is initial age state, and we're
+    // the first person in, reset it to the original position.  (ie, prop the default state
+    // we've got right now)
+    if (pos && pos->fZ < kMaxNegativeZPos && initialSync) {
+        SimLog("Physical {} loaded out of range state.  Forcing initial state to server.", GetKeyName());
+        DirtySynchState(kSDLPhysical, plSynchedObject::kBCastToClients);
+        return;
+    }
+
+    ISetPoseSim(pos, rot, wakeup);
+    if (linV)
+        SetLinearVelocitySim(*linV, wakeup);
+    if (angV)
+        SetAngularVelocitySim(*angV, wakeup);
+
+    SendNewLocation(false, true);
+}
+
+// ==========================================================================
+
+void plPXPhysical::SendNewLocation(bool synchTransform, bool isSynchUpdate)
+{
+    // Called after the simulation has run....sends new positions to the various scene objects
+    // *** want to do this in response to an update message....
+    if (CanSynchPosition(isSynchUpdate)) {
+        plProfile_Inc(MaySendLocation);
+
+        if (!GetProperty(plSimulationInterface::kPassive)) {
+            hsMatrix44 curl2w = fCachedLocal2World;
+            // we're going to cache the transform before sending so we can recognize if it comes back
+            IGetTransformGlobal(fCachedLocal2World);
+
+            if (!curl2w.Compare(fCachedLocal2World, .0001f)) {
+                plProfile_Inc(LocationsSent);
+                plProfile_BeginLap(PhysicsUpdates, GetKeyName().c_str());
+
+                if (fCachedLocal2World.GetTranslate().fZ < kMaxNegativeZPos)
+                {
+                    SimLog("Physical {} fell to {.1f} ({.1f} is the max).  Suppressing.", GetKeyName(), fCachedLocal2World.GetTranslate().fZ, kMaxNegativeZPos);
+                    // Since this has probably been falling for a while, and thus not getting any syncs,
+                    // make sure to save it's current pos so we'll know to reset it later
+                    DirtySynchState(kSDLPhysical, plSynchedObject::kBCastToClients);
+                    IEnable(false);
+                }
+
+                hsMatrix44 w2l;
+                fCachedLocal2World.GetInverse(&w2l);
+                plCorrectionMsg *pCorrMsg = new plCorrectionMsg(GetObjectKey(), fCachedLocal2World, w2l, synchTransform);
```

Comment:
error: cannot bind non-const lvalue reference of type ‘plKey&’ to an rvalue of type ‘plKey’

<pre><font color="#FFFFFF"><b>/home/dpogue/Coding/Plasma/Sources/Plasma/PubUtilLib/plPhysX/plGenericPhysical.cpp:</b></font> In member function ‘<font color="#FFFFFF"><b>virtual void plPXPhysical::SendNewLocation(bool, bool)</b></font>’:
<font color="#FFFFFF"><b>/home/dpogue/Coding/Plasma/Sources/Plasma/PubUtilLib/plPhysX/plGenericPhysical.cpp:488:77:</b></font> <font color="#EF2929"><b>error: </b></font>cannot bind non-const lvalue reference of type ‘<font color="#FFFFFF"><b>plKey&amp;</b></font>’ to an rvalue of type ‘<font color="#FFFFFF"><b>plKey</b></font>’
  488 |                 plCorrectionMsg *pCorrMsg = new plCorrectionMsg(<font color="#EF2929"><b>GetObjectKey()</b></font>, fCachedLocal2World, w2l, synchTransform);
      |                                                                 <font color="#EF2929"><b>~~~~~~~~~~~~^~</b></font>
In file included from <font color="#FFFFFF"><b>/home/dpogue/Coding/Plasma/Sources/Plasma/PubUtilLib/plPhysX/plGenericPhysical.cpp:55</b></font>:
<font color="#FFFFFF"><b>/home/dpogue/Coding/Plasma/Sources/Plasma/PubUtilLib/plPhysX/../../NucleusLib/pnMessage/plCorrectionMsg.h:54:28:</b></font> <font color="#34E2E2"><b>note: </b></font>  initializing argument 1 of ‘<font color="#FFFFFF"><b>plCorrectionMsg::plCorrectionMsg(plKey&amp;, const hsMatrix44&amp;, const hsMatrix44&amp;, bool)</b></font>’
   54 |     plCorrectionMsg(<font color="#34E2E2"><b>plKey&amp; r</b></font>, const hsMatrix44&amp; l2w, const hsMatrix44&amp; w2l,
      |                     <font color="#34E2E2"><b>~~~~~~~^</b></font>
</pre>